dev/publish_halyard.py

```diff
@@ -0,0 +1,194 @@
+# to a running ssh-agent on the machine this script will run on:
+#
+# > <copy or rsync the key to the vm>
+# > eval `ssh-agent`
+# > ssh-add ~/.ssh/<key with access to github repo>
+#
+# If you are running this script on Jenkins, you can configure Jenkins to handle SSH credentials.
+
+import argparse
+import os
+import sys
+import yaml
+
+from annotate_source import Annotator
+from spinnaker.run import check_run_quick
+from build_release import BackgroundProcess
+
+
+class HalyardPublisher(object):
+  """Publishes a nightly version of Halyard to be a stable version.
+  """
+
+  def __init__(self, options, build_number=None):
+    self.__annotator = None
+    self.__build_number = build_number or os.environ.get('BUILD_NUMBER')
+    self.__github_publisher = options.github_publisher
+    self.__hal_nightly_bucket_uri = options.hal_nightly_bucket_uri
+    self.__halyard_repo_uri = options.halyard_repo_uri
+    self.__nightly_version = options.nightly_version
+    self.__options = options
+    self.__patch_release = options.patch_release
+    self.__stable_branch = ''
+    self.__stable_version = options.stable_version
+    self.__stable_version_tag = ''
+
+  def __checkout_halyard_repo(self):
+    """Clones the Halyard git repo at the commit at which we built the nightly version.
+    """
+    bucket_uri = self.__hal_nightly_bucket_uri
+    local_bucket_name = os.path.basename(bucket_uri)
+    print 'cloning: {0}'.format(self.__halyard_repo_uri)
+    check_run_quick('git clone {0}'.format(self.__halyard_repo_uri))
+
+    # Read the Halyard nightly bucket file.
+    print ('Fetching Halyard nightly build info from {0}. Writing locally to {1}.'
+           .format(bucket_uri, local_bucket_name))
+    if not os.path.exists(local_bucket_name):
+        os.mkdir(local_bucket_name)
+    check_run_quick('gsutil rsync -r -d {remote_uri} {local_bucket}'
+                    .format(remote_uri=bucket_uri, local_bucket=local_bucket_name))
+    nightly_commit_file = '{0}/nightly-version-commits.yml'.format(local_bucket_name)
+    nightly_commit_dict = {}
+    with open(nightly_commit_file, 'r') as ncf:
+      nightly_commit_dict = yaml.load(ncf.read())
+
+    # Check Halyard out at the correct commit.
+    commit_to_build = nightly_commit_dict.get(self.__nightly_version, None)
+    if not commit_to_build:
+      raise ValueError('No commit hash recorded in {bucket} for Halyard nightly version {nightly}, exiting.'
+                       .format(bucket=bucket_uri, nightly=self.__nightly_version))
+    print ('Checking out Halyard from {0} repo at commit {1}.'
+           .format(self.__halyard_repo_uri, commit_to_build))
+    check_run_quick('git -C halyard checkout {0}'.format(commit_to_build))
+
+  def __tag_halyard_repo(self):
+    """Tags the Halyard repo with a tag derived from --stable_version.
+    """
+    self.__stable_version_tag = self.__stable_version
+    if not self.__stable_version_tag.startswith('version-'):
```

Comment:
Does this mean I don't have to pass `version-1.0.0` to the jenkins job, and `1.0.0` is fine?